### PR TITLE
Configure header_checks and sender_canonical_maps.

### PR DESCRIPTION
**Pull request by mtyaka**
#### Background

This patch modifies the postfix_queue role to add ability to configure postfix `header_checks` and `sender_canonical_maps`.

The contents of header_checks file is controlled by the `POSTFIX_QUEUE_HEADER_CHECKS` ansible variable and sender_canonical_maps by the `POSTFIX_QUEUE_SENDER_CANONICAL_MAPS` variable.
Both are optional and default to blank strings (empty configuration files).
#### Testing

Set `POSTFIX_QUEUE_HEADER_CHECKS` and `POSTFIX_QUEUE_SENDER_CANONICAL_MAPS` to desired values and run the `postfix_queue` role, for example:

```
ansible-playbook -i "<host-ip>," playbooks/run_role.yml -u ubuntu -e "role=postfix_queue" \
    -e '{"POSTFIX_QUEUE_HEADER_CHECKS": "/^From:(.*)$/   PREPEND Reply-To:$1"}' \
    -e '{"POSTFIX_QUEUE_SENDER_CANONICAL_MAPS": "user@email.com rewritten@email.com"}'
```

Check that `/etc/postfix/header_checks` and `/etc/postifx/sender_canonical_maps` on the target host contain expected content and that `/etc/postfix/main.cf` contains the following lines:

```
sender_canonical_maps = hash:/etc/postfix/sender_canonical_maps
header_checks = regexp:/etc/postfix/header_checks
```

Then run playbook again without setting the vars:

```
ansible-playbook -i "<host-ip>," playbooks/run_role.yml -u ubuntu -e "role=postfix_queue"
```

Check that `/etc/postfix/header_checks` and `/etc/postfix/sender_canonical_maps` files don't contain any configuration.

---

**JIRA ticket**: https://openedx.atlassian.net/browse/OSPR-1301

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @mtyaka! It looks like you're a member of a company that does contract work for edX. If you're doing this work as part of a paid contract with edX, you should talk to edX about who will review this pull request. If this work is not part of a paid contract with edX, then you should ensure that there is an OSPR issue to track this work in [JIRA](https://openedx.atlassian.net), so that we don't lose track of your pull request.

To automatically create an OSPR issue for this pull request, just visit this link: https://openedx-webhooks.herokuapp.com/github/process_pr?repo=edx%2Fconfiguration&number=3124

File: playbooks/roles/postfix_queue/tasks/main.yml [hunk 1]
```diff
@@ -45,3 +47,28 @@
   command: "postmap hash:{{ postfix_queue_password_file }}"
   when: postfix_queue_password.changed
   notify: restart postfix
+
+- name: Configure postfix sender canonical maps
+  copy: >
+    dest="{{ postfix_queue_sender_canonical_maps_file }}"
+    content="# configured by ansible:\n{{ POSTFIX_QUEUE_SENDER_CANONICAL_MAPS }}"
```
Comment (itsjeyd) [inline playbooks/roles/postfix_queue/tasks/main.yml]: @mtyaka Nit: "Configured by Ansible", perhaps? If you adjust it, there's a couple other places in this file that will also need to be updated (L33 and L43).

Comment (mtyaka) [inline playbooks/roles/postfix_queue/tasks/main.yml]: @itsjeyd Fixed, thanks.

**Comment by itsjeyd** [comment]
@mtyaka :+1:

**Comment by edx-webhook** [comment]
Thanks for the pull request, @mtyaka! I've created [OSPR-1301](https://openedx.atlassian.net/browse/OSPR-1301) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).

**Comment by feanil** [comment]
Changes look good to me.  @fredsmith can you do the second review on this.

**Comment by e0d** [comment]
@mtyaka can you update this role to use new style task formatting for ansible 2.0 support?  Simple exemplar, https://github.com/edx/configuration/pull/3240/files.

 I'd prefer to have an explicit restart and not use handlers unless there's a specific reason it make sense for this role.  Are updates made via this role that don't restart?

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @mtyaka! It looks like you're a member of a company that does contract work for edX. If you're doing this work as part of a paid contract with edX, you should talk to edX about who will review this pull request. If this work is not part of a paid contract with edX, then you should ensure that there is an OSPR issue to track this work in [JIRA](https://openedx.atlassian.net), so that we don't lose track of your pull request.

To automatically create an OSPR issue for this pull request, just visit this link: https://openedx-webhooks.herokuapp.com/github/process_pr?repo=edx%2Fconfiguration&number=3124

**Comment by haikuginger** [comment]
@e0d, I've made those changes for @mtyaka; he's unavailable at the moment, so just @ me with whatever else you might need on this PR.

**Comment by e0d** [comment]
@haikuginger I think we're good to merge this :+1:

**Pull request merged**